pkg/reconciler/ingress/lister.go

```diff
@@ -82,9 +86,21 @@ func (l *gatewayPodTargetLister) getIngressUrls(ing *v1alpha1.Ingress, gatewayIp
 				target.URLs = domainsToURL(domains, scheme)
 			}
 		} else {
+			ns, err := l.kubeclient.CoreV1().Namespaces().Get(context.TODO(), ing.Namespace, metav1.GetOptions{})
```

Comment:
Again, concerned this might be doing a GET to the apiserver for each probe, rather than caching the namespace annotations. If it's not, that's great. If it is, it seems like we should pass a NamespacesLister like the EndpointsLister.

Reply:
Agree make sense in this context. Updating...